File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/Readonly/Question.cs
using MaterialAdvisor.Application.Models.Shared;
using MaterialAdvisor.Data.Enums;

namespace MaterialAdvisor.Application.Models.Readonly;

public class Question
{
    public byte Number { get; set; }

    public byte Version { get; set; }

    public double Points { get; set; }

    public QuestionType Type { get; set; }

    public IEnumerable<LanguageText> Texts { get; set; } = null!;

    public IEnumerable<AnswerGroup> AnswerGroups { get; set; } = [];
}

File: src/app/material-advisor/src/app/features/topic/models/Question.ts
import { LanguageText } from "@shared/models/LanguageText";
import { QuestionType } from "@shared/types/QuestionEnum";
import { AnswerGroupModel } from "./AnswerGroup";

export interface QuestionModel {
  number: number;
  points: number;
  type: QuestionType;
  content: LanguageText[];
  answerGroups: AnswerGroupModel[];
}
